reana_workflow_controller/rest/utils.py

```diff
@@ -135,11 +136,8 @@ def _start_workflow_db(workflow, parameters):
 
 def stop_workflow(workflow):
     """Stop a given workflow."""
-    if workflow.status == RunStatus.running:
-        kwrm = KubernetesWorkflowRunManager(workflow)
-        kwrm.stop_batch_workflow_run()
-        workflow.status = RunStatus.stopped
-        Session.add(workflow)
+    if workflow.can_transition_to(RunStatus.stopped):
+        _update_workflow_status(workflow, RunStatus.stopped, logs="")
         Session.commit()
```

Comment:
I guess calling `Session.commit()` here is unnecessary, since it is already being called by [`_update_workflow_status`](https://github.com/mdonadoni/reana-workflow-controller/blob/e1f8b39f37e528ea762f7caa24a275927d58aee7/reana_workflow_controller/consumer.py#L143) ([here](https://github.com/reanahub/reana-db/blob/cc6052242fd14cf3413b793d0aa32a24871fe1b1/reana_db/models.py#L755) the `reana_db` code), right?

Reply:
It is still needed because even though the status is commited by `Workflow.update_workflow_status`, the logs are fetched and saved to the database after that:
https://github.com/reanahub/reana-workflow-controller/blob/817b019b3745862436e99570c10c6d8ea35533f4/reana_workflow_controller/consumer.py#L143-L156